spark-bigquery-dsv2/spark-3.1-bigquery-lib/src/main/java/com/google/cloud/spark/bigquery/v2/Spark31BigQueryScanBuilder.java

```diff
@@ -46,8 +48,10 @@
         SupportsReportStatistics,
         SupportsQueryPushdown {
 
-  private BigQueryDataSourceReaderContext ctx;
-  private InputPartition[] partitions;
+  protected final Logger logger = LoggerFactory.getLogger(getClass());
```

Comment:
It is used in the Spark 3.2 subclass. I took a a page from the Spark framework book where the superclass defines the logger for the hierarchy, but for each class the logger name is of that class (because it is not a static variable)